tests/net/igmp/src/main.c

```diff
@@ -559,4 +607,58 @@ ZTEST_USER(net_igmp, test_socket_catch_join_with_index)
 	socket_leave_group_with_index(&my_addr);
 }
 
+static void igmp_send_query(bool is_imgpv3)
+{
+	struct net_pkt *pkt;
+
+	is_report_sent = false;
+	is_join_msg_ok = false;
+
+	is_igmpv2_query_sent = false;
+	is_igmpv3_query_sent = false;
+
+	/* Joining group first to get reply on query*/
+	join_group();
+
+	is_igmpv2_query_sent = !is_imgpv3;
+	is_igmpv3_query_sent = is_imgpv3;
+
+	pkt = prepare_igmp_query(net_iface, is_imgpv3);
+	if (!pkt) {
+		zassert_true(false, "IGMPv2 query packet prep failed");
+	}
+
+	if (net_ipv4_input(pkt, false)) {
+		net_pkt_unref(pkt);
+		zassert_true(false, "Failed to send");
+	}
```

Comment:
It's already unexpected to fail, asserting would abort anyhow.

```suggestion
	zassert_equal(net_ipv4_input(pkt, false), NET_OK, "Failed to send");
```